data-gathering/bot.py

```diff
@@ -1,726 +1,529 @@
 
+   sum_of_warnings = tmp_to_check + tmp_synoptic + tmp_help + tmp_correct + tmp_curiosity + tmp_divide + tmp_sources + tmp_localism + tmp_pov
+   sum_of_warnings += tmp_nn + tmp_recentism + tmp_manual_style + tmp_translation + tmp_wikificare + tmp_stub + tmp_stub_section + tmp_copy_control
 
 
+   tmp_without_sources = sum(t.count(template) for template in language_config.get("without_sources"))
+   tmp_to_clarify = sum(t.count(template) for template in language_config.get("clarify"))
 
-# extract a specific template from a given text. returns either the modified text (if a template is found) or the original text
-def findtemplate(text):
+   return str(sum_of_warnings), str(tmp_without_sources), str(tmp_to_clarify)
 
-      #  copies of the input text with the first two characters removed.
-      tmp = text[2:]
 
-      tmp2 = text[2:]
 
-      # include only the portion of text from the beginning to the first occurrence of "}}" plus two characters.
-      tmp = tmp[:tmp.find("}}")+2]
 
-      if "{{" in tmp:
+# This function is designed to iteratively find and extract templates from a text until no more templates are present
+def find_template(text):
 
-         # substring of tmp starting from the first occurrence of "{{" to the end.
-          tmp3 = tmp[tmp.find("{{"):]
+   tmp = text[2:]
+   tmp2 = text[2:]
+   tmp = tmp[:tmp.find("}}")+2]
 
-         # replace tmp3 with a placeholder string 
-          tmp2 = tmp2.replace(tmp3,"$$$$$$$$$$$$$$")
+   if "{{" in tmp:
 
+      tmp3 = tmp[tmp.find("{{"):]
+      tmp2 = tmp2.replace(tmp3,"$$$$$$$$$$$$$$")
 
-         # include only the portion up to the first occurrence of "}}" plus two characters
 
-          tmp2 = tmp2[:tmp2.find("}}")+2]
-         
-         # placeholder is replaced back with the original template (tmp3)
-          tmp2 = tmp2.replace("$$$$$$$$$$$$$$",tmp3)
+      tmp2 = tmp2[:tmp2.find("}}")+2]
+      tmp2 = tmp2.replace("$$$$$$$$$$$$$$",tmp3)
 
-          return tmp2
+      return tmp2
+   return tmp
 
-      return tmp
 
 
 
-
-# calculate the length of an introductory part of a text while handling various text manipulations.
-def lengthIncipit(text):
-
+# Function to calculate the length of the introduction
+#Incipit means the opening of a manuscript, early printed book. Hence, incipit == introduction 
+def calculate_introduction_length(text):
    incipit = text
-
-   #  extract the part of the text up to the first occurrence of "\n==" using incipit = incipit[:incipit.find("\n==")].
-
    incipit = incipit[:incipit.find("\n==")]
+   template_count  =incipit.count('{{')
 
-   ntemplate  = incipit.count('{{')
-
-   incipitclear = incipit
-
-   # Handling '{{formatnum:' Templates:
-   fn = incipit.count("{{formatnum:")
-
-   for i in range(fn):
+   # incipitclear = incipit
+   format_num = incipit.count("{{formatnum:")
 
+   for i in range(format_num):
       tmp = incipit[incipit.find("{{formatnum:"):]
```

Comment:
just to make sure, is it "{{formatnum:" the same in all languages?

Reply:
I think it is. I had to do some research to be sure. I came across this: <https://www.wikidata.org/wiki/Q14940601>